### PR TITLE
TG-138: LogIn Page

### PR DESCRIPTION
**Pull request by fernandocarbone1**
_LogIn page made by Martín Pueblas and Fernando Carboné_

1. Form with email & password inputs
2. Validations
3. Styles

### CODE REVIEW

File: src/Components/EmployeesFlow/validations.js [hunk 1]
```diff
@@ -132,3 +132,25 @@ export const employeeValidationUpdate = Joi.object({
     }),
   active: Joi.boolean().sensitive()
 });
+
+export const employeeValidationLogIn = Joi.object({
+  email: Joi.string()
+    .required('This field is required')
```
Comment (FefeTroanes) [inline src/Components/EmployeesFlow/validations.js:138]: Hace falta esto aca? Ya esta en la linea 145. Es consulta

File: src/Components/EmployeesFlow/validations.js [hunk 2]
```diff
@@ -132,3 +132,25 @@ export const employeeValidationUpdate = Joi.object({
...
+    .min(2)
+    .regex(/^\w+([.-]?\w+)@\w+([.-]?\w+)(.\w{2,3})+$/)
+    .lowercase()
+    .messages({
+      'string.min': 'Email must contain at least 2 characters',
+      'string.pattern.base': 'Invalid email format',
+      'string.empty': 'This field is required'
+    }),
+  password: Joi.string()
+    .required('This field is required')
```
Comment (FefeTroanes) [inline src/Components/EmployeesFlow/validations.js:148]: Lo mismo aca con la linea 154

File: src/Components/Login/index.js [hunk 3]
```diff
@@ -1,7 +1,110 @@
...
+//     },
+//     [history]
+//   );
+
+//   const { currentUser } = useContext(AuthContext);
+
+//   if (currentUser) {
+//     return <Redirect to="/" />;
+//   }
+// };
```
Comment (FefeTroanes) [inline src/Components/Login/index.js:32]: Todo esto que esta comentado que onda? Es consulta, no se si quedo para usarse despues

File: src/Components/Login/index.js [hunk 4]
```diff
@@ -1,7 +1,110 @@
...
+
+const Login = () => {
+  const history = useHistory();
+  const message = useSelector((state) => state.employees.message);
+  const dispatch = useDispatch();
+  const [showModalMessage, setShowModalMessage] = useState(false);
+  const handleCloseMessage = () => {
+    setShowModalMessage(false);
+    setShowModalMessage('');
```
Comment (FefeTroanes) [inline src/Components/Login/index.js:41]: Esto hace falta que este asi? Si lo pones vacio directamente funciona igual? Es consulta

File: src/Components/Login/login.module.css [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    transition: all 0.35s;
+  }
+  .buttonContinue:hover::after, .buttonReset:hover::after, .buttonOk:hover::after{
+    left: 0;
+    right: 0;
+    top: 0;
+    bottom: 0;
+    -webkit-transition: all 0.35s;
+    transition: all 0.35s;
+  };
```
Comment (FefeTroanes) [inline src/Components/Login/login.module.css:142]: Aca github pide una linea en blanco al final, re gede

**Review by FefeTroanes — COMMENTED** [review]
👌🏼

File: src/helper/firebase/index.js [hunk 6]
```diff
@@ -22,7 +21,11 @@ export const tokenListener = () => {
       const {
         claims: { role }
       } = await user.getIdTokenResult();
+      sessionStorage.setItem('token', token);
       store.dispatch(setAuthentication({ token, role }));
+    } else {
+      sessionStorage.removeItem('token');
+      store.dispatch(setAuthentication());
```
Comment (nicolascostanza) [inline src/helper/firebase/index.js:28]: Deberia guardarse en el localstorage, para una mejor experiencia de usuario

**Comment by nicolascostanza** [comment]
image

**Comment by nicolascostanza** [comment]
image
Acá si ingresas email o password invalida no debería salir modal, nomas un mensaje con usuario y/o contraseña incorrecta

**Comment by nicolascostanza** [comment]
Además si pongo el mail y contraseña que dejo Martin tampoco puedo ingresar

**Comment by nicolascostanza** [comment]
El screen con estilos deberia quedar asi, en una columna, cambiar el color rosado por el azul usado en la paleta de colores. El button reset no va 
image

**Review by nicolascostanza — DISMISSED** [review]

**Review by ilcosme — APPROVED** [review]

**Comment by Clarosluciano** [comment]
# Test result:
Functionalities: fail

# Bugs found:
1. When clicks on cotinue button nothing happens. This happens with every entity:
image
image

**Comment by Clarosluciano** [comment]
# Bugs Found:
1. When entering the admin data, it is redirected to a blank page:
image
image

2. When entering with the employee data and clicking on "go home" the user is redirected to timesheets. When the "edit profile" button is clicked the user is redirected to another employee's profile:
image
image
image
When the "go home" button of this new profile is pressed, the "edit profile" button disappears from the table:
image
image
4. Subsequently, the button will not appear in any table:
image
image
image
image
5. By clicking on admin or super admin the user will be redirected to the initial profile with which they entered the employee section. And if you click on "go home" the cycle will start again:
image
image

File: src/Components/EmployeesFlow/validations.js (hunk 2 above)
Comment (martinalbrnz) [reply src/Components/EmployeesFlow/validations.js:148]: fixed

File: src/Components/EmployeesFlow/validations.js (hunk 1 above)
Comment (martinalbrnz) [reply src/Components/EmployeesFlow/validations.js:138]: fixed jeje

**Comment by Clarosluciano** [comment]
# Bugs found:
1. When the button "edit profile" is in the page, the buttons of CRUD does nothing:
image

**Review by Clarosluciano — APPROVED** [review]

**Review by FefeTroanes — APPROVED** [review]
😎👉🏼

**Pull request merged**